MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/AdsImaSDKListener.kt

```diff
@@ -71,15 +73,21 @@ class AdsImaSDKListener private constructor(
    * @param event, event to be updated.
    * @param ad, current ad event that is being processed.
    */
-  private fun setupAdViewData(event: PlaybackEvent, ad: Ad?) {
+  private fun setupAdViewData(event: MuxAdEvent, ad: Ad?) {
     val viewData = ViewData()
+    val adData = AdData();
```

Comment:
Would it make sense to define a constructor for `AdData` that takes in an `Ad`? Because then this logic:

```
        ad.adId?.let { adData.adId = it }
        ad.creativeId?.let { adData.adCreativeId = it }
        ad.universalAdIdValue?.let { adData.adUniversalId = it }
```

can be moved into the constructor / initializer and becomes more testable AISI.

I'm taking the opportunity to build context on this codebase (and Kotlin more generally) via this question so feel free to disregard if there's something I'm missing or address at some more opportune time! :)

Reply:
Unfortunately `AdData` is not directly coupled to the IMA SDK. The Java and Objc Cores make no assumptions about player or ad framework at all.

I'd rather write all our "data" classes differently than they're written, but this is the pattern that the Java core uses. It's "fine" Kotlin, but it could be more idiomatic. My preference would be for `AdData` to be a `data class` (something like a `struct`) that is immutable and updated by copy.. (very) Unfortunately, the underlying core SDKs depend on them being mutable and shared between many objects.